src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -594,31 +614,113 @@ class FunctionDictGetNoType final : public IFunction
 
 private:
 
+    std::pair<ColumnPtr, ColumnPtr> getDefaultsShortCircuit(
+        IColumn::Filter && default_mask,
+        const DataTypePtr & result_type,
+        const ColumnWithTypeAndName & last_argument) const
+    {
+        auto rows = default_mask.size();
+
+        auto mask_col = ColumnUInt8::create();
+        mask_col->getData() = std::move(default_mask);
+        ColumnPtr mask_col_res = std::move(mask_col);
+
+        IColumn::Filter mask(rows, 1);
+        auto mask_info = extractMask(mask, mask_col_res);
+        ColumnWithTypeAndName column_before_cast = last_argument;
+        maskedExecute(column_before_cast, mask, mask_info);
```

Comment:
I think we can avoid calling `maskedExecute` now, just use `maskedExecute` with initial `default_mask`. In this case `mask_info` will be unknown, but we can just initialize it with general case `has_ones = true, has_zeros = true`.

Reply:
Now I understand why switch 0 and 1 at first. Added default value and comment in maskedExecute, and further simplified logic of getDefaultsShortCircuit.